src/System.Management.Automation/engine/InternalCommands.cs

```diff
@@ -351,80 +386,107 @@ protected override void ProcessRecord()
-                        }
-                        else
-                        {
-                            // member is a method
-                            if (member is PSMethodInfo)
+                            // first we check if the member is a ParameterizedProperty
+                            PSParameterizedProperty targetParameterizedProperty = member as PSParameterizedProperty;
+                            if (targetParameterizedProperty != null)
                             {
-                                // first we check if the member is a ParameterizedProperty
-                                PSParameterizedProperty targetParameterizedProperty = member as PSParameterizedProperty;
-                                if (targetParameterizedProperty != null)
-                                {
-                                    // should process
-                                    string propertyAction = String.Format(CultureInfo.InvariantCulture,
-                                        InternalCommandStrings.ForEachObjectPropertyAction, targetParameterizedProperty.Name);
+                                // should process
+                                string propertyAction = String.Format(CultureInfo.InvariantCulture,
+                                    InternalCommandStrings.ForEachObjectPropertyAction, targetParameterizedProperty.Name);
 
-                                    // ParameterizedProperty always take parameters, so we output the member.Value directly
-                                    if (ShouldProcess(_targetString, propertyAction))
-                                    {
-                                        WriteObject(member.Value);
-                                    }
-                                    return;
+                                // ParameterizedProperty always take parameters, so we output the member.Value directly
+                                if (ShouldProcess(_targetString, propertyAction))
+                                {
+                                    WriteObject(member.Value);
                                 }
+                                return;
+                            }
 
-                                PSMethodInfo targetMethod = member as PSMethodInfo;
-                                Dbg.Assert(targetMethod != null, "targetMethod should not be null here.");
-                                try
-                                {
-                                    // should process
-                                    string methodAction = String.Format(CultureInfo.InvariantCulture,
-                                        InternalCommandStrings.ForEachObjectMethodActionWithoutArguments, targetMethod.Name);
+                            PSMethodInfo targetMethod = member as PSMethodInfo;
+                            Dbg.Assert(targetMethod != null, "targetMethod should not be null here.");
```

Comment:
Maybe throw here with user friendly message then in line 413 with standard null exception?

Reply:
I'm not going to change this; it's totally unrelated to my change.

Reply:
I leave this open - we should fix this in the PR or later.